flict/flictlib/license.py

```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
-                license_list = lep
-            else:
-                logger.license_logger.debug("daisy: " + str(license_list))
-                logger.license_logger.debug("daisy: " + str(lep))
-                logger.license_logger.debug(
-                    " -------------====== Andy ====----------------- SINGLE: " + str(license_list))
-                new_list = []
-                for item in license_list:
-                    logger.license_logger.debug(
-                        "  item: " + str(item) + " <--- " + str(lep))
-                    for lep_item in lep:
-                        logger.license_logger.debug(
-                            "    item: " + str(item) + " <--- " + str(lep_item))
-                        new_item = list(set(item + lep_item))
-                        logger.license_logger.debug("    item: " + str(item))
-                        new_list.append(new_item)
-                    logger.license_logger.debug("    list: " + str(new_list))
-                license_list = new_list
-            return license_list
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
+
+        # single license
+        if len(license_list) == 0:
+            return lep
+
+        new_list = []
+        for item in license_list:
+            for lep_item in lep:
+                new_item = list(set(item + lep_item))
+                new_list.append(new_item)
+
+        return new_list
 
     def _manage_list_item_or(self, license_list, lep):
-        logger.license_logger.debug(" * Orleans: " + (str(lep)))
         if isinstance(lep, LicenseExpressionList):
-            raise Exception("Internal failure. Wrong type " +
-                            str(type(lep)) + " for: " + str(lep))
-        else:
-            # single license
-            logger.license_logger.debug("HERE I AM .... \"" + str(lep) + "\"")
-            if len(license_list) == 0:
-                new_list = lep
-                logger.license_logger.debug(
-                    "topsss: " + str(license_list) + " size: " + str(len(license_list)))
-                logger.license_logger.debug(
-                    "topsss: " + str(lep) + " size: " + str(len(lep)))
-                logger.license_logger.debug(
-                    "topsss: " + str(new_list) + " size: " + str(len(new_list)))
-            else:
-                new_list = license_list
-                logger.license_logger.debug("dapsss: " + str(license_list))
-                logger.license_logger.debug("dappss: " + str(lep))
-                for lep_item in lep:
-                    logger.license_logger.debug(
-                        "    item: " + str(license_list) + " <--- " + str(lep_item))
-                    new_list.append(lep_item)
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
+
+        # single license
+        if len(license_list) == 0:
+            return lep
+
+        new_list = license_list
```

Comment:
I guess a `return list(lep)` could be used for this whole block here

Reply:
But wouldn't that miss out on the potential content of `license_list`?

Reply:
I gave it a spin and it fails a bunch of tests.

Reply:
True so it is `return list(license_list) + list(lep)` then - maybe also worth making it unique by doing `return list(set(list(license_list) + list(lep)))`

Reply:
Just more errors.

Reply:
in this case - leave it for now as it is